corehq/apps/reports/standard/sms.py

```diff
@@ -489,8 +489,18 @@ def get_contact_link(couch_recipient, couch_recipient_doc_type, raw=False):
             ] if val != Ellipsis]
 
     def _get_message_event_display(self, message, content_cache):
+        event = None
         if message.messaging_subevent:
             event = message.messaging_subevent.parent
+        elif message.xforms_session_couch_id:
+            try:
+                subevent = (MessagingSubEvent.objects
+                            .get(parent__domain=self.domain,
+                                 xforms_session__couch_id=message.xforms_session_couch_id))
```

Comment:
This will raise `MultipleObjectsReturned` if there is more than one, but it doesn't look like that's handled here. Does this query always return a single result or none at all?

Reply:
Great question.  I assumed that there isn't more than one `MessagingSubEvent`s for a given `SQLXFormSession`, but I don't actually know that to be true.  Do you know?  I think I'd rather try it out and see (under the FF, of course), than premptively handle something that might not occur in practice, thus further muddying the waters.  I'll look through the code a bit and see if I can find any hints.

Reply:
Sounds like a good plan. Might be a good idea to proactively monitore for errors as this gets released to broader audiences.